backend/inspirehep/records/api/base.py

```diff
@@ -363,6 +366,22 @@ def redirect_pids(self, pids):
                 self.redirect_pid(pid_type, pid_value)
             return pids
 
+    @classmethod
+    def delete_records_from_deleted_records(cls, data):
+        # Hack for migrator in case new record takes pids from other records
+        # which should be deleted but they are not deleted yet.
+        for pid in data.get("deleted_records", []):
+            pid_type, pid_value = PidStoreBase.get_pid_from_record_uri(pid["$ref"])
+            try:
+                record_to_delete = cls.get_record_by_pid_value(
+                    pid_value, pid_type, original_record=True
+                )
+            except PIDDoesNotExistError:
+                # Pid is not registered!
```

Comment:
do we need this comment? :)

Reply:
Comment is not needed, but I added a log as this is potential error/bug so we want to have some kind of trace that this situation happened.